purrr/slack_cmd.py

```diff
@@ -0,0 +1,145 @@
+"""Handler for Slack slash-command events."""
+
+import data_helper
+import slack_helper
+
+
+slack = slack_helper.shared_client
+
+
+def on_slack_slash_command(event):
+    """Entry-point for AutoKitteh sessions triggered by Slack slash-command events.
+
+    - /purrr help
+    - /purrr opt-in
+    - /purrr opt-out
+    - /purrr list
+    - /purrr status [PR]
+    - /purrr approve [PR]
+
+    About slash commands: https://api.slack.com/interactivity/slash-commands
+    See also: https://api.slack.com/interactivity/handling#message_responses
+
+    Args:
+        event: Slack event data.
+    """
+    # Split the command into normalized arguments.
+    data = event.data
+    args = str(data.text).lower().split()
+
+    # Route further processing to the appropriate command handler.
+    if not args or "help" in args:
+        _help(data, args)
+        return
+
+    for cmd, func, _ in _COMMANDS:
+        if cmd == args[0]:
+            func(data, args)
+            return
```

Comment:
Insertion order preservation has been an official feature in Python dictionaries only since https://docs.python.org/3.7/whatsnew/3.7.html, that's why I'm not used to it. But it's a good idea - done.